black_hole/discord.py

```diff
@@ -152,13 +173,46 @@ async def _send_all(self):
         """Send all pending webhook messages."""
         log.debug("working on %d jobs...", len(self._queue))
         for job in self._queue:
+            # key used to store the message
+            store_key = (job["author_jid"], job["upstream_message_id"])
+
+            # key used to lookup the message (as the reply message has a different id,
+            # using upstream_message_id directly would always yield non-hits to the
+            # message id store)
+            lookup_key = (job["author_jid"], job["reply_message_id"])
+            webhook_url = job["webhook_url"]
+            resp = None
+
             try:
-                resp = await self.session.post(job["webhook_url"], json=job["payload"])
-                await asyncio.sleep(self.config["discord"].get("delay", 0.25))
+                try:
+                    discord_message_id = self._message_id_store[lookup_key]
+                    resp = await self.session.patch(
+                        f"{webhook_url}/messages/{discord_message_id}",
+                        json=job["payload"],
+                        params={"wait": "true"},
+                    )
+                except KeyError:
```

Comment:
this works but i'd prefer a `if lookup_key in self._message_id_store` check instead of using exceptions so that the reader sees the branch taken most first in the code